### PR TITLE
Add metric for FTP

### PR DESCRIPTION
**Pull request by srijan-27**
- Added `app_ftp_stats` histogram metric for FTP with labels as `type` and `status`
- Fixed formatting in logs
- Closes: #892

**Checklist:**

- [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
- [x] All new code is covered by unit tests.
- [x] This PR does not decrease the overall code coverage.
- [x] I have reviewed the code comments and documentation for clarity.

### CODE REVIEW

File: pkg/gofr/datasource/file/ftp/fs.go [hunk 1]
```diff
@@ -76,6 +77,9 @@ func (f *fileSystem) UseMetrics(metrics interface{}) {
 func (f *fileSystem) Connect() {
 	var status string
 
+	ftpBuckets := []float64{.05, .075, .1, .125, .15, .2, .3, .5, .75, 1, 2, 3, 4, 5, 7.5, 10}
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/ftp/fs.go:80]: Is this bucket made for ftp only? I mean maybe there could be a default bucket values to reuse by defining a variable in a package?

This is more a question than a blocking comment

File: pkg/gofr/datasource/file/ftp/fs_test.go [hunk 2]
```diff
@@ -501,7 +509,8 @@ func TestMkDirAll(t *testing.T) {
 	mockLogger.EXPECT().Logf(gomock.Any()).AnyTimes()
 	mockLogger.EXPECT().Errorf(gomock.Any(), gomock.Any()).AnyTimes()
 	mockLogger.EXPECT().Debug(gomock.Any()).AnyTimes()
-	mockMetrics.EXPECT().RecordHistogram(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+	mockMetrics.EXPECT().RecordHistogram(gomock.Any(), "app_ftp_stats", gomock.Any(),
```
Comment (ccoVeille) [inline pkg/gofr/datasource/file/ftp/fs_test.go:512]: Maybe a constant for "app_ftp_stats"

Something shared between code and tests

**Comment by ccoVeille** [comment]
Also, maybe you could address this unresolved comment made on a previous PR

- https://github.com/gofr-dev/gofr/pull/957#discussion_r1734511563

File: pkg/gofr/datasource/file/ftp/fs.go (hunk 1 above)
Comment (srijan-27) [reply pkg/gofr/datasource/file/ftp/fs.go:80]: yeah, the thought process behind making separate buckets (although same value) for each datasource is that in future, we can manipulate buckets to achieve higher precision for data.

**Review by ccoVeille — APPROVED** [review]

**Review by aryanmehrotra — APPROVED** [review]

**Review by coolwednesday — APPROVED** [review]

**Pull request merged**